lib/streamlit/Report.py

```diff
@@ -53,7 +53,7 @@ def get_url(cls, host_ip):
         port = _get_browser_address_bar_port()
         return "http://%(host_ip)s:%(port)s" % {"host_ip": host_ip, "port": port}
 
-    def __init__(self, script_path, argv):
+    def __init__(self, script_path, argv, command_line):
```

Comment:
Is `argv` still needed at all? From a quick look at the code I think you can remove it, along with all places that depend on it — all the way down to `sys.argv = argv` in ScriptRunner.py.

Reply:
The problem is that `ScriptRunner.py` does not simply execute a command line, but it parses and recompile the script because of "magic" and then executes in this way:
```
            with modified_sys_path(self._report), self._set_execing_flag():
                exec(code, module.__dict__)
```

Reply:
In a way it is handy to have `args` and `script path` separately

Reply:
Ok, agreed on a way to do this per private conversation.